crypto/evp/evp_fetch.c

```diff
@@ -282,21 +317,37 @@ inner_evp_generic_fetch(OPENSSL_CTX *libctx, int operation_id,
             ossl_method_store_cache_set(store, meth_id, properties, method,
                                         up_ref_method, free_method);
         }
+
+        /*
+         * If we never were in the constructor, the algorithm to be fetched
+         * is unsupported.
+         */
+        unsupported = !mcmdata.flag_construct_error_occured;
     }
 
-    return method;
-}
+ end:
+    if (method == NULL) {
+        int code =
+            unsupported ? EVP_R_UNSUPPORTED_ALGORITHM : EVP_R_FETCH_FAILED;
 
-#ifndef FIPS_MODULE
-static const char *libctx_descriptor(OPENSSL_CTX *libctx)
-{
-    if (openssl_ctx_is_global_default(libctx))
-        return "Global default library context";
-    if (openssl_ctx_is_default(libctx))
-        return "Thread-local default library context";
-    return "Non-default library context";
-}
+#ifdef FIPS_MODULE
+        ERR_raise(ERR_LIB_EVP, code);
```

Comment:
The entire point of #12855 was to include more information for this error.  It *must* include the algorithm being fetched and ideally the property query too.

Reply:
So are you saying that I should incorporate #12855 into this PR?  I didn't want to steal away your glory, ya know...

Reply:
Do it.  I don't need the glory.